config/initializers/rails_performance.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_performance'
+
+Rails.application.config.after_initialize do
+  if Settings.rails_performance.enabled
+    RailsPerformance::RailsPerformanceController.class_eval do
+      include SessionHandler
+      before_action :check_user_not_authorized
+
+      # Modify CSP for Performance Gem
+      content_security_policy do |p|
+        p.style_src :self, "'unsafe-inline'"
+        p.script_src_elem :self,
+                          "'unsafe-inline'",
+                          'https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/js/all.min.js',
+                          'https://code.highcharts.com'
+      end
+
+      protected
+
+      # Helper to use SessionHandler to check for permissions
+      def check_user_not_authorized
+        unless real_user&.admin_user?
+          render 'shared/http_status',
+                 formats: [:html], locals: { code: '403', message: HttpStatusHelper::ERROR_CODE['message']['403'] },
+                 status: :forbidden, layout: false
+        end
+      end
+    end
+  end
+
+  if defined?(RailsPerformance)
+    RailsPerformance.setup do |config|
+      config.enabled = Settings.rails_performance.enabled
+      config.redis = Redis::Namespace.new(Rails.root.to_s)
```

Comment:
1. Let's go with the default name (`Redis::Namespace.new("#{Rails.env}-rails-performance")`)
2. I think we need to pass in the MarkUs Redis instance as the `redis` keyword argument (as illustrated in the documentation). But, we can't yet assume the MarkUs Redis instance has been initialized when we run this file. Instead, add this specific configuration setting to the `redis.rb` initializer and use `Redis.current` as an argument to the `Redis::Namespace.new` (we do something similar with `Resque.redis` in that file already).